plugins/default-browser-emulator/injected-scripts/_proxyUtils.ts

```diff
@@ -192,97 +128,112 @@ function proxyConstructor<T, K extends keyof T>(
   const toString = descriptor.value.toString();
   descriptor.value = new Proxy(descriptor.value, {
     construct() {
-      try {
-        const result = overrideFn(...arguments);
-        if (result !== ProxyOverride.callOriginal) {
-          return result as any;
-        }
-      } catch (err) {
-        throw cleanErrorStack(err);
-      }
-      try {
-        return ReflectCached.construct(...arguments);
-      } catch (err) {
-        throw cleanErrorStack(err, { stripStartingReflect: true });
+      const result = overrideFn(...arguments);
+      if (result !== ProxyOverride.callOriginal) {
+        return result as any;
       }
+
+      return ReflectCached.construct(...arguments);
     },
   });
   overriddenFns.set(descriptor.value, toString);
   ObjectCached.defineProperty(owner, key, descriptor);
 }
 
-function internalCreateFnProxy(
-  targetFn: any,
-  descriptor: PropertyDescriptor,
-  onApply: (target: any, thisArg: any, argArray: any[]) => any,
-) {
-  const toString = targetFn.toString();
-  const proxy = new Proxy<any>(targetFn, {
-    apply: onApply,
-    setPrototypeOf(target: any, newPrototype: any): boolean {
-      let protoTarget = newPrototype;
-      let newPrototypeProto;
-      try {
-        newPrototypeProto = newPrototype?.__proto__;
-      } catch {}
-      if (newPrototype === proxy || newPrototypeProto === proxy) {
+const setProtoTracker = new WeakSet<Error>();
+
+function internalCreateFnProxy<T extends object>(opts: {
+  target: T;
+  descriptor?: any;
+  custom?: ProxyHandler<T>;
+  inner?: ProxyHandler<T> & { disableGetProxyOnFunction?: boolean };
+  disableStoreToString?: boolean;
+}) {
+  function apply(target: any, thisArg: any, argArray: any[]) {
+    if (opts.inner?.apply) {
+      return opts.inner.apply(target, thisArg, argArray);
+    }
+    return ReflectCached.apply(target, thisArg, argArray);
+  }
+
+  function setPrototypeOf(target: any, newPrototype: any): boolean {
+    let protoTarget = newPrototype;
+    let newPrototypeProto;
+    try {
+      newPrototypeProto = Object.getPrototypeOf(newPrototype);
+    } catch {}
+    if (newPrototype === proxy || newPrototypeProto === proxy) {
+      protoTarget = target;
+    }
+
+    const temp = { stack: 'TODO' };
```

Comment:
Can we throw a different word in here so it doesn't autocomplete to project "todos"?